common/validators.js

```diff
@@ -97,5 +97,42 @@
                 // });
             }
         };
+    })
+
+    .directive('timestamp', function() {
+        return {
+            restrict: 'A',
```

Comment:
Yeah but `<input> ... </input>` is not valid HTML.

From my understanding of the Angular [docs](https://docs.angularjs.org/guide/directive), attribute would be the preferred practice here, since this new directive is not meant to be used alone.
>**When should I use an attribute versus an element?** Use an element when you are creating a component that is in control of the template. The common case for this is when you are creating a Domain-Specific Language for parts of your template. Use an attribute when you are decorating an existing element with new functionality.